File: Framework/LanguageSupport/java/MMIJava/src/Adapter/IMMUInstantiation.java
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Andreas Kaiser, Felix Gaisbauer

package Adapter;


import de.mosim.mmi.mmu.MMUDescription;

public interface IMMUInstantiation {
    /*
    Interface for a Java MMU instantiator
     */

    MotionModelUnitBase InstantiateMMU(MMUDescription mmuDescription);
}

File: Framework/LanguageSupport/cs/MMICSharp/Adapter/IMMUInstantiation.cs
// SPDX-License-Identifier: MIT
// The content of this file has been developed in the context of the MOSIM research project.
// Original author(s): Felix Gaisbauer

using MMICSharp.Common;
using MMIStandard;

namespace MMICSharp.Adapter
{
    /// <summary>
    /// Interface for encapsulating different MMU Instantiation methods
    /// </summary>
    public interface IMMUInstantiation
    {
        IMotionModelUnitDev InstantiateMMU(MMULoadingProperty loadingProperty);
    }
}
